core/domain/base_model_validators.py

```diff
@@ -176,12 +200,18 @@ def _validate_model_domain_object_instances(cls, item):
             item: ndb.Model. Entity to validate.
         """
         try:
-            model_domain_object_instance = (
+            domain_object = (
                 cls._get_model_domain_object_instance(item))
-            if model_domain_object_instance is None:
+            if domain_object is None:
                 # No domain object exists for this storage model class.
                 return
-            model_domain_object_instance.validate()
+            validation_type = cls._get_domain_object_validation_type(item)
+            if validation_type == VALIDATION_MODE_NEUTRAL:
+                domain_object.validate()
+            elif validation_type == VALIDATION_MODE_STRICT:
+                domain_object.validate(strict=True)
+            else:
```

Comment:
Do elif NON_STRICT, and then "else raise Exception".

Reply:
Done.